File: frontend/src/types/scan.ts
export interface Vulnerability {
  id: number;
  scan_id: number;
  vulnerability_type: string;
  file_path: string | null;
  line_number: number | null;
  count: number | null;
  severity: string | null;
  cvss: number | null;
  description: string | null;
  recommendation: string | null;
  created_at: string | null;
}

export interface Scan {
  id: number;
  project_id: number;
  status: string;
  repository_url: string | null;
  branch: string | null;
  started_at: string | null;
  completed_at: string | null;
  files_scanned: number | null;
  vulnerabilities_found: number | null;
  risk_score: number | null;
  created_at: string | null;
  vulnerabilities: Vulnerability[];
}

export interface ScanCreate {
  branch: string;
}

File: backend/app/schemas/scan.py
from pydantic import BaseModel
from datetime import datetime

from app.schemas.vulnerability import VulnerabilityResponse


class ScanCreate(BaseModel):
    branch: str = "main"


class ScanResponse(BaseModel):
    id: int
    project_id: int
    status: str

    repository_url: str | None = None
    branch: str | None = None

    started_at: datetime | None = None
    completed_at: datetime | None = None

    files_scanned: int | None = None
    vulnerabilities_found: int | None = None
    risk_score: float | None = None

    created_at: datetime | None = None

    vulnerabilities: list[VulnerabilityResponse] = []

    class Config:
        from_attributes = True